app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -176,3 +184,34 @@ class CiYaml {
     }
   }
 }
+
+/// Class to expose the PinnedVersion method for testing.
+class DependencyValidator {
+  /// Dependency is guaranteed to be non empty as it must be found before this
+  /// method is called.
+  ///
+  /// Checks a dependency string for a pinned version.
+  /// If a version is found then it must not be empty or 'latest.'
+  static void hasVersion({required String dependencyJsonString}) {
+    final List<String> exceptions = <String>[];
+    // decoded will contain a list of maps
+    dynamic decoded = json.decode(dependencyJsonString);
+
+    for (Map<String, dynamic> depMap in decoded) {
+    
+      if (!depMap.containsKey('version')) {
```

Comment:
Do we need a skip list, like `xcode`?

Reply:
Can you please clarify here? Do you mean we skip certain dependencies in this check or is it something else?

Reply:
Yeah, I meant if we need to skip certain dependencies in this check, such as `xcode`, whose dependency version would not be counted (we rely on property value instead).
After a rethink, I think it would be good to keep all dependencies consistent with a version.
Let's keep the current logic as you have right now.